### ISSUE TITLE
CameraRoll.getPhotos does not return any image but useCameraRoll.getPhotos does [iOS]

### ISSUE THREAD

**Issue by hotaryuzaki** [react-native-cameraroll/react-native-cameraroll#581]
The issue in iOS only
 
```
const params = {
        assetType: "All",
        groupTypes: "All",
        first: limit,
        after: offset,
        groupName: "",
        include: ['fileSize', 'filename', 'imageSize', 'playableDuration'],
};
const getPhotos = await CameraRoll.getPhotos(params);
```

and the return is
```
 {"edges": [], "limited": false, "page_info": {"has_next_page": false}}
```

i tried change assetType, groupNam, groupTypes, remove limit and offset, but nothing
then i tried useCameraRoll.getPhotos() the return is

```
[
  {
    "node": {
      "group_name": "All Photos",
      "image": [Object
      ],
      "location": null,
      "modificationTimestamp": 1704687227.4046373,
      "subTypes": [Array
      ],
      "timestamp": 1704687227,
      "type": "image"
    }
  },
  {
     ...
  }
]
```

what is wrong?? is there any setup in iOS that i missed?
and how to set params like in CameraRoll for useCameraRoll??

@react-native-camera-roll/camera-roll: 5.10.0
react-native: 0.71.14

**Comment by hotaryuzaki** [comment]
is upgraded but the issue still persists
@react-native-camera-roll/camera-roll: 7.4.0

**Comment by hotaryuzaki** [comment]
i found that the "after" object is causing this issue.
so in iOS we cannot use after?
how to create pagination with fromTime and toTime?? i think this will make the number of results different on each page.

**Comment by Oleksandr073** [comment]
Hello! This is incredible but I'm just now trying to use this library on iOS and I'm having the same problem! Indeed, photos are not displayed when "after" is specified. And how then to make pagination? Display only first N photos on iOS? Is there an alternative to this library? Or use the default iOS gallery?

**Comment by Oleksandr073** [comment]
@hotaryuzaki, I solved the problem. The 'after' parameter must have the value 'end_cursor'. On Android, this value is the number of photos, but as a string '0', '50', and so on. On iOS, this value, as I understand it, is the file name. The point is to pass the 'end_cursor' value to 'after', and the initial value should be undefined.

```tsx
export const MediaGallery = () => {
  const [galleryPage, setGalleryPage] = useState<string | undefined>();
  const [endCursor, setEndCursor] = useState<string | undefined>();
  const [mediaFiles, setMediaFiles] = useState<PhotoIdentifier[]>([]);

  useEffect(() => {
    CameraRoll.getPhotos({
      first: 50,
      after: galleryPage,
      assetType: 'All',
      include: ['playableDuration'],
    })
      .then(({ edges, page_info: { has_next_page, end_cursor } }) => {
        setMediaFiles((prevFiles) => [...prevFiles, ...edges]);
        if (has_next_page && end_cursor) {
          setEndCursor(end_cursor);
        }
      })
      .catch(() => {
        setEndCursor(undefined);
        setGalleryPage(undefined);
        setMediaFiles([]);
      });
  }, [galleryPage]);

  const onPaginate = () => {
    setGalleryPage(endCursor);
  };

  return (
    <FlatList
      data={mediaFiles}
      onEndReachedThreshold={0.5}
      onEndReached={onPaginate}
      keyExtractor={(item) => item.node.image.uri}
      renderItem={({
        item: {
          node: {
            image: { uri, playableDuration },
          },
        },
      }) => (
        <View className="w-full h-full">
          <Image className="w-full h-full" source={{ uri }} />
          {playableDuration && (
            <View className="absolute right-1 bottom-1">
              <Tag
                label={secondsToMediaDurationFormat(playableDuration)}
                size="sm"
              />
            </View>
          )}
        </View>
      )}
      numColumns={3}
      columnWrapperStyle={{
        columnGap: GAP,
        marginBottom: GAP,
      }}
    />
  );
};
```

**Comment by Arjan-Zuidema** [comment]
Hi @hotaryuzaki and @Oleksandr073 

Sorry for the late reaction.

It is described in the docs that you should indeed use the `end_cursor` as @Oleksandr073 just commented above.

I will close this now.

**Comment by hotaryuzaki** [comment]
> @hotaryuzaki, I solved the problem. The 'after' parameter must have the value 'end_cursor'. On Android, this value is the number of photos, but as a string '0', '50', and so on. On iOS, this value, as I understand it, is the file name. The point is to pass the 'end_cursor' value to 'after', and the initial value should be undefined.
> 
> ```tsx
> export const MediaGallery = () => {
>   const [galleryPage, setGalleryPage] = useState<string | undefined>();
>   const [endCursor, setEndCursor] = useState<string | undefined>();
>   const [mediaFiles, setMediaFiles] = useState<PhotoIdentifier[]>([]);
> 
>   useEffect(() => {
>     CameraRoll.getPhotos({
>       first: 50,
>       after: galleryPage,
>       assetType: 'All',
>       include: ['playableDuration'],
>     })
>       .then(({ edges, page_info: { has_next_page, end_cursor } }) => {
>         setMediaFiles((prevFiles) => [...prevFiles, ...edges]);
>         if (has_next_page && end_cursor) {
>           setEndCursor(end_cursor);
>         }
>       })
>       .catch(() => {
>         setEndCursor(undefined);
>         setGalleryPage(undefined);
>         setMediaFiles([]);
>       });
>   }, [galleryPage]);
> 
>   const onPaginate = () => {
>     setGalleryPage(endCursor);
>   };
> 
>   return (
>     <FlatList
>       data={mediaFiles}
>       onEndReachedThreshold={0.5}
>       onEndReached={onPaginate}
>       keyExtractor={(item) => item.node.image.uri}
>       renderItem={({
>         item: {
>           node: {
>             image: { uri, playableDuration },
>           },
>         },
>       }) => (
>         <View style={styles.mediaItemWrapper}>
>           <Image style={styles.mediaItemPhoto} source={{ uri }} />
>           {playableDuration && (
>             <View style={styles.mediaItemDuration}>
>               <Tag
>                 label={secondsToMediaDurationFormat(playableDuration)}
>                 size="sm"
>               />
>             </View>
>           )}
>         </View>
>       )}
>     />
>   );
> };
> ```

Last Thursday I found that too, so on Android the "after" string is a number but on iOS is like the ID of the last image.
It makes me confused because I thought it would be the same.
So I have to change the logic like this
```
...offset && { after: offset }
```

But I have another issue with getAlbums, some iPhone return nothing, it doesn't get the Recents smartAlbum.
But maybe I will create another issue next week.

Thank you

**Comment by Arjan-Zuidema** [comment]
Yes, please create an other issue for different issues 😉
